src/on_pull_request.ts

```diff
@@ -60,47 +63,60 @@ async function byPassChecks(octokit, issue, sha, checks) {
  * label has been applied. In the case that either have, the corresponding
  * check will be removed and recorded.
  */
-async function onLabel(octokit: github.GitHub, context, input: Input) {
-  const { issue, payload } = context;
-  const { number } = issue;
-  const { html_url } = payload.pull_request;
+async function onLabel(octokit: GitHubClient, context: Context, input: Input) {
+  const { issue, pull_request, label } = context.payload;
+  const { owner, repo, number } = context.issue;
+
+  core.debug(`label event received: ${pp(context.issue)}`);
 
-  core.debug(`label event received: ${pp(payload)}`);
-  if (payload.label.name === input.skipCILabel) {
+  if (label.name === input.skipCILabel && pull_request) {
     core.debug(`skip_ci_label applied`);
 
-    await postMessage(`Bypassing CI checks for <${html_url}|#${number}>`);
+    await postMessage(`Bypassing CI checks for <${pull_request.html_url}|#${number}>`);
 
     await comment(
       octokit,
       issue,
-      `Bypassing CI checks - ${payload.label.name} applied`
+      `Bypassing CI checks - ${label.name} applied`
     );
 
     await byPassChecks(
       octokit,
-      issue,
-      payload.pull_request.head.sha,
+      pull_request,
+      pull_request.head.sha,
       input.requiredChecks,
     );
   }
 
-  if (payload.label.name === input.skipApprovalLabel) {
+  if (label.name === input.skipApprovalLabel && pull_request) {
     core.debug(`skip_approval applied`);
 
-    await postMessage(`Bypassing peer approval for <${html_url}|#${number}>`);
+    await postMessage(`Bypassing peer approval for <${pull_request.html_url}|#${number}>`);
 
     await octokit.pulls.createReview({
-      owner: issue.owner,
-      repo: issue.repo,
-      pull_number: issue.number,
-      body: `Skipping approval check - ${payload.label.name} applied`,
+      owner,
+      repo,
+      pull_number: number,
+      body: `Skipping approval check - ${label.name} applied`,
       event: 'APPROVE',
     });
   }
+
+  if (label.name === input.posthocApprovalLabel) {
+    const author = (issue?.user?.id || pull_request?.user?.id);
```

Comment:
This _may_ be able to be simplified to just
```suggestion
    const author = context.issue.user.id;
```

Reply:
I sooooooooooo wish. `issue` on context is a special getter that just returns repo info. For user, you have to dig into the payload issue/pr, which have slightly different signatures. This is why I tried to lean more into their types a bit, because these payloads are more than a little opaque.